### ISSUE TITLE
Proposal: add `EnforcedStyle: indent_when` to Layout/CaseIndentation

### ISSUE THREAD

**Issue by searls** [rubocop-hq/rubocop#6447]
Similar to the (default) `indent` style of Layout/AccessModifierIndentation, it strikes me as odd that there's no RuboCop setting for indenting `when` underneath `case`, since `when` clauses are subordinate to the `case` keyword in much the same way that `private` is subordinate to `class`.

``` ruby
# The `indent_when` style enforces that `when` clauses be
# indented one step further than `case` and end

# bad
case fruit
when :apple then "🍎"
when :banana then "🍌"
end

# good
case fruit
  when :apple then "🍎"
  when :banana then "🍌"
end
```

It's possible this option should(?) only be applied to one-liner `when` statements. Interested in folks' thoughts on that.

**Comment by mikegee** [comment]
This should do the trick for you:

```yaml
Layout/CaseIndentation:
  IndentOneStep: true
```

**Comment by searls** [comment]
Awesome! Thank you, @mikegee!

**Comment by bbatsov** [comment]
I'm shocked by how inconsistent some of our configuration options have become. 😆

**Comment by searls** [comment]
While I have everyone's attention: there's not a way to limit this rule based on whether the `when` statement is single line or multiline, is there?

For single line clauses, I much prefer this:

```ruby
case fruit
  when :apple then "🍎"
  when :banana then "🍌"
end
```

But would not for multi-line `when` blocks (not that I'd encourage ever having those)

**Comment by mikegee** [comment]
I don't see options for that in [the docs](https://rubocop.readthedocs.io/en/latest/cops_layout/#layoutcaseindentation).

**Comment by pjscopeland** [comment]
@searls I'd like to see this option as well.

```yaml
# Supports --auto-correct
Layout/CaseIndentation:
  IndentOneStep: with_thens
```

```ruby
case owner
when 'I'
  attrs[:user] = @current_user
when 'they'
  attrs[:user] = @it
end
```

```ruby
case owner
  when 'I' then attrs[:user] = @current_user
  when 'they' then attrs[:user] = @it
end
```
